File: src/lib/classes/SpotifyImage.ts
export class SpotifyImage {
  height?: number;
  url!: string;
  width?: number;

  private constructor() {
  }

  public static fromImageObject(image: SpotifyApi.ImageObject): SpotifyImage {
    const spotifyImage = new SpotifyImage();
    spotifyImage.height = image.height;
    spotifyImage.url = image.url;
    spotifyImage.width = image.width;
    return spotifyImage;
  }
}

File: src/main/java/de/wittenbude/expandify/models/spotifydata/helper/SpotifyImage.java
package de.wittenbude.expandify.models.spotifydata.helper;

import lombok.Data;
import lombok.NoArgsConstructor;
import se.michaelthelin.spotify.model_objects.specification.Image;

@Data
@NoArgsConstructor
public class SpotifyImage {
    private Integer height;
    private String url;
    private Integer width;

    public SpotifyImage(Image image) {
        this.height = image.getHeight();
        this.url = image.getUrl();
        this.width = image.getWidth();
    }
}
